precompiles/staking/approve_test.go

```diff
@@ -316,28 +317,28 @@ func (s *PrecompileTestSuite) TestApprove() {
 			func(data []byte, inputArgs []interface{}) {
 				s.Require().Equal(data, cmn.TrueValue)
 
-				authz, expirationTime := s.CheckAuthorization(staking.DelegateAuthz, s.address, s.address)
+				authz, expirationTime := CheckAuthorizationWithContext(ctx, s.network.App.AuthzKeeper, staking.DelegateAuthz, s.keyring.GetAddr(0), s.keyring.GetAddr(0))
 				s.Require().NotNil(authz)
 				s.Require().NotNil(expirationTime)
 				s.Require().Equal(authz.AuthorizationType, staking.DelegateAuthz)
 				s.Require().Equal(authz.MaxTokens, &sdk.Coin{Denom: s.bondDenom, Amount: math.NewInt(1e18)})
 
-				authz, expirationTime = s.CheckAuthorization(staking.UndelegateAuthz, s.address, s.address)
+				authz, expirationTime = CheckAuthorizationWithContext(ctx, s.network.App.AuthzKeeper, staking.UndelegateAuthz, s.keyring.GetAddr(0), s.keyring.GetAddr(0))
 				s.Require().NotNil(authz)
 				s.Require().NotNil(expirationTime)
 
 				s.Require().Equal(authz.AuthorizationType, staking.UndelegateAuthz)
 				s.Require().Equal(authz.MaxTokens, &sdk.Coin{Denom: s.bondDenom, Amount: math.NewInt(1e18)})
 
-				authz, expirationTime = s.CheckAuthorization(staking.RedelegateAuthz, s.address, s.address)
+				authz, expirationTime = CheckAuthorizationWithContext(ctx, s.network.App.AuthzKeeper, staking.RedelegateAuthz, s.keyring.GetAddr(0), s.keyring.GetAddr(0))
 				s.Require().NotNil(authz)
 				s.Require().NotNil(expirationTime)
 
 				s.Require().Equal(authz.AuthorizationType, staking.RedelegateAuthz)
 				s.Require().Equal(authz.MaxTokens, &sdk.Coin{Denom: s.bondDenom, Amount: math.NewInt(1e18)})
 
 				// TODO: Bug here it returns 3 REDELEGATE authorizations
-				allAuthz, err := s.app.AuthzKeeper.GetAuthorizations(s.ctx, s.address.Bytes(), s.address.Bytes())
+				allAuthz, err := s.network.App.AuthzKeeper.GetAuthorizations(s.network.GetContext(), s.keyring.GetAddr(0).Bytes(), s.keyring.GetAddr(0).Bytes())
```

Comment:
unnecessary conversion


```suggestion
				allAuthz, err := s.network.App.AuthzKeeper.GetAuthorizations(s.network.GetContext(), s.keyring.GetAccAddr(0), s.keyring.GetAccAddr(0))
```